### ISSUE TITLE
Doesn't compile

### ISSUE THREAD

**Issue by GoogleCodeExporter** [TASVideos/vba-rerecording#34]
```
I am using top-of-tree as of this post.

What steps will reproduce the problem?
1. ./configure
2. make
3. Note numerous error messages

What is the expected output? What do you see instead?
I expect it to compile.  It doesn't.

1.
There are multiple places, such as
src/proc/proc.cpp
src/sdl/debugger.cpp
that try to cast pointers to u32.
That isn't even valid and isn't a good idea to begin with.
On AMD64, pointers are 64-bit, not 32.  Are you sure
you even should be casting?

2.
There seems to be confusion over the use of bool vs. bool8 in
src/Sound.(h|cpp)
Changing all of them into plain bools seems to fix it on my machine.
Though, I don't know what it's trying to do to begin with,
especially since bool is native in C++.

3.
It depends on zutil.h which clearly states:

/* WARNING: this file should *not* be used by applications. It is
   part of the implementation of the compression library and is
   subject to change. Applications should only use zlib.h.
 */

4.
Then, it fails to link:
/home/scubed/software/vba-1.7.2-21/src/gb/gbMemory.cpp:326: undefined
reference to `VBAMovieGetFrameCounter()'
/home/scubed/software/vba-1.7.2-21/src/gb/gbMemory.cpp:325: undefined
reference to `VBAMovieLoading()'
...
I don't know the cause of this one.
It is trying to link in ../gb/libgb.a
which appears to have the symbol.

In short, the code needs massive cleaning.


What version of the product are you using? On what operating system?
Linux on AMD64
```

Original issue reported on code.google.com by `scubed.g...@lingcog.iit.edu` on 6 Jun 2009 at 4:40
